File: Seminars/src/main/java/org/example/SeminarOOP5/hw/model/Cat.java
package org.example.SeminarOOP5.hw.model;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
/**
 * Класс Cat расширяет класс Animal дополнительными полями, доступными для котов
 */
public class Cat extends Animal {
    private int age;

    public Cat(String name, int age) {
        super(name);
        this.age = age;
    }

    @Override
    public String toString() { return String.format("Кличка кота: %s, Возраст: %d", super.getName(), age); }
}

File: Seminars/src/main/java/org/example/SeminarOOP7/hw/Cat.py
from Animal import Animal

class Cat(Animal):
    def __init__(self, nickname, age):
        super().__init__(nickname)
        self._age = age

    @property
    def age(self):
        return self._age

    @age.setter
    def age(self, new_age):
        self._age = new_age

    def __str__(self) -> str:
        return f"{self.nickname} {self._age}"